lang/src/org/partiql/lang/eval/ExprValueExtensions.kt

```diff
@@ -299,6 +299,12 @@ fun ExprValue.cast(
                         return valueFactory.newDate(LocalDate.of(ts.year, ts.month, ts.day))
                     }
                     type.isText -> try {
+                        val datePatternRegex = Regex("\\d\\d\\d\\d-\\d\\d-\\d\\d")
```

Comment:
Minor--this probably should be constructed once and used here.

Reply:
Do you mean moving this as top level value? It's been constructed and used only once here.

Reply:
I think @almann means that a single instance should be constructed once outside of this function (i.e. at the top level in the file) and used here.

Reply:
Done. Thanks.